### PR TITLE
Remove need for profile changes in storage OAuth support

### PR DESCRIPTION
**Pull request by markcowl**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by markcowl** [comment]
signing run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/623/

**Comment by blueww** [comment]
@markcowl 

The change looks fail in the sign job with error: 

> 01:15:28   Publish-PSArtifactUtility : PowerShellGet cannot resolve the module dependency 
> 01:15:28   'AzureRM.Profile' of the module 'AzureRM' on the repository 
> 01:15:28   '0ae7ab7d-0adb-4f9f-b05b-28ef883f51bc'. Verify that the dependent module 
> 01:15:28   'AzureRM.Profile' is available in the repository 
> 01:15:28   '0ae7ab7d-0adb-4f9f-b05b-28ef883f51bc'. If this dependent module 
> 01:15:28   'AzureRM.Profile' is managed externally, add it to the 
> 01:15:28   ExternalModuleDependencies entry in the PSData section of the module manifest.
> 01:15:28   At C:\Program 
> 01:15:28   Files\WindowsPowerShell\Modules\PowerShellGet\1.6.0\PSModule.psm1:1447 char:17
> 01:15:28   + ...             Publish-PSArtifactUtility @PublishPSArtifactUtility_Param ...
> 01:15:28   +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 01:15:28       + CategoryInfo          : InvalidOperation: (:) [Publish-PSArtifactUtility 
> 01:15:28      ], InvalidOperationException
> 01:15:28       + FullyQualifiedErrorId : UnableToResolveModuleDependency,Publish-PSArtifa 
> 01:15:28      ctUtility
>

**Comment by markcowl** [comment]
@blueww The sign failure was a machine-related issue in compiling the msi, the bits there are fine.

Will take a look at the issue you report.  My suspicion is that there is an issue with module layout on your machine, but will take a look.  I can't really test without an enabled subscription for this feature, do you have one?

**Comment by markcowl** [comment]
signing run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/630/

File: tools/AzureRM/AzureRM.psd1 [hunk 1]
```diff
@@ -52,7 +52,6 @@ CLRVersion = '4.0'
 
 # Modules that must be imported into the global environment prior to importing this module
 RequiredModules = @(@{ModuleName = 'AzureRM.Profile'; RequiredVersion = '5.0.0'; }, 
-               @{ModuleName = 'Azure.Storage'; RequiredVersion = '4.3.0'; },
```
Comment (praries880) [inline tools/AzureRM/AzureRM.psd1]: why this change? shouldn't we bump this to 4.4.0 instead?
